File: arrays/217-contains-duplicate/solution.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        containsDict = {}

        for x in nums:
            if x in containsDict:
                return True
            containsDict[x] = 1
        return False

        # Add this block to run the code locally in VS Code
if __name__ == "__main__":
    solver = Solution()
    # Replace with the test cases from the LeetCode problem
    result = solver.containsDuplicate([1,2,3,1])
    print(f"Output: {result}")

File: arrays/217-contains-duplicate/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    
    const containsMap = new Map();
    for(let x of nums){

        if(containsMap.has(x)) {
            return true;
        }
        containsMap.set(x, 1);
    }
    return false;
};


console.log(containsDuplicate([1,2,3,1]));
console.log(containsDuplicate([1,2,3,4]));